### ISSUE TITLE
ij.py.from_java() returns scyjava.JavaList when input is java.util.ArrayList

### ISSUE THREAD

**Issue by carshadi** [imagej/pyimagej#100]
```python
import imagej
ij = imagej.init(fiji_path, headless=False)
from scyjava import jimport
ArrayList = jimport('java.util.ArrayList')
arrayList = ArrayList()
print(type(arrayList))  # <java class 'java.util.ArrayList'>
print(type(ij.py.from_java(arrayList)))  # <class 'scyjava.JavaList'>
```
I'm not sure if this is the intended behavior, but the [Usage.md](https://github.com/imagej/pyimagej/blob/master/doc/Usage.md#converting-between-java-and-python) states the return type should be a Python `list`

Running `pyimagej 1.0.0` installed via conda-forge with `openjdk 8.0.265` in a jupyter notebook

**Comment by ctrueden** [comment]
@carshadi Thanks for the report. It's confusing because when converting collections from Python to Java, data is copied. But when converting collections from Java to Python, the Java objects are wrapped using Python duck typing. So the type `scyjava.JavaList` is a Python class wrapping some Java object implementing `java.util.List`. Technically, the `scyjava.JavaList` object _isn't_ a Python `list`, but it _has_ all the same attributes and functions, which means that from a Pythonic perspective, it can be _used_ as a `list` without issue.

That said, I believe there are a couple of gaps in the implemented functions of the Python collection wrappers. Not sure about `JavaList`, but I remember running into an issue recently with `JavaSet`, where some set functions from newer versions of Python were missing from the wrapper.

If you really need a true `list`, I was going to say you could just do a full slice using `[:]`, but it seems there a bug in `JavaList` with that:
```python
>>> jl = ArrayList([1, 3, 5, 7, 9])
>>> jl.toString()
'[1, 3, 5, 7, 9]'
>>> pl = scyjava.to_python(jl)
>>> type(pl)
<class 'scyjava.JavaList'>
>>> print(pl)
[1, 3, 5, 7, 9]
>>> pl[:]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/curtis/miniconda3/envs/i2k-2020-pyimagej/lib/python3.8/site-packages/scyjava/__init__.py", line 330, in __getitem__
    return to_python(self.jobj.get(key), gentle=True)
TypeError: No matching overloads found for java.util.ArrayList.get(slice), options are:
	public java.lang.Object java.util.ArrayList.get(int)
```
While this bug needs to be fixed (I filed scijava/scyjava#24 for it), it's often the case that you don't need the `scyjava.to_python` a.k.a. `ij.py.from_java` function at all anymore, now that we're using JPype, because JPype has a lot more magic than PyJNIus does. E.g.:
```python
>>> import scyjava
>>> def process_list(l):
...   for i in l:
...     print(i + i)
... 
>>> process_list([1, 2, 3, 4])
2
4
6
8
>>> ArrayList = scyjava.jimport('java.util.ArrayList')
>>> jl = ArrayList([1, 2, 3, 4])
>>> process_list(jl)
2
4
6
8
>>> jl = ArrayList(['a', 'b', 'c', 'd'])
>>> process_list(jl)
aa
bb
cc
dd
>>> process_list(['a', 'b', 'c', 'd'])
aa
bb
cc
dd
>>> jl + jl
<java object 'java.util.ArrayList'>
>>> print(jl + jl)
[a, b, c, d, a, b, c, d]
```
Still, let us know if there are specific conversions you need that aren't working.

**Comment by carshadi** [comment]
thanks, this is helpful. casting with `list()` is just fine for us
